libs/core/src/lib/skeleton/components/skeleton.component.ts

```diff
@@ -0,0 +1,73 @@
+import { ChangeDetectionStrategy, Component, HostBinding, Input, ViewEncapsulation } from '@angular/core';
+import { random } from 'lodash-es';
+
+export type SkeletonType = 'circle' | 'rectangle' | 'text';
+export type SkeletonWidth = 'rand' | string;
+export type SkeletonHeight = 'auto' | string;
+
+@Component({
+    selector: 'fd-skeleton',
+    template: `<ng-content></ng-content>`,
+    styleUrls: ['skeleton.component.scss'],
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    // eslint-disable-next-line @angular-eslint/no-host-metadata-property
+    host: {
+        class: 'fd-skeleton'
+    }
+})
+export class SkeletonComponent {
+    /** Wight of the skeleton */
+    @Input()
+    set width(value: SkeletonWidth) {
+        if (value === 'rand') {
+            this._width = getRandomWidth();
+            return;
+        }
+
+        this._width = value;
+    }
+
+    /** Height of the skeleton. Relevant if type 'circle' or 'rectangle'. */
+    @Input()
+    height: SkeletonHeight;
+
+    /** Type of the skeleton. When set to 'text' height is automatically calculated. */
+    @Input()
+    type: SkeletonType = 'rectangle';
+
+    @Input()
+    @HostBinding('class.fd-skeleton--inherit')
+    inheritSize = false;
+
+    /** Whether the skeleton has animation. */
+    @Input()
+    @HostBinding('class.fd-skeleton--animated')
+    animation = true;
+
+    /** @hidden */
+    @HostBinding('style.width')
+    _width: string;
+
+    /** @hidden */
+    @HostBinding('style.height')
+    get _height(): string {
+        return this.type === 'text' ? 'auto' : this.height;
```

Comment:
Following the above comment, maybe add a fallback to use width if height is not provided at all so it would draw perfect circle or square by just providing width OR height